cmd/controller/main.go

```diff
@@ -217,13 +217,22 @@ func main() {
 	gsSetController := gameserversets.NewController(wh, health, gsCounter,
 		kubeClient, extClient, agonesClient, agonesInformerFactory)
 	fleetController := fleets.NewController(wh, health, kubeClient, extClient, agonesClient, agonesInformerFactory)
-	gasController := gameserverallocations.NewController(api, health, gsCounter, kubeClient, kubeInformerFactory,
-		agonesClient, agonesInformerFactory, 10*time.Second, 30*time.Second, ctlConf.AllocationBatchWaitTime)
 	fasController := fleetautoscalers.NewController(wh, health,
 		kubeClient, extClient, agonesClient, agonesInformerFactory)
 
+	if !runtime.FeatureEnabled(runtime.FeatureSplitControllerAndExtensions) {
+		gameservers.NewExtensions(wh)
```

Comment:
I'd originally written
> Maybe this should happen later (since it requires more untangling) but I don't understand why `wh` needs to be passed to all of the control loops. It seems like they shouldn't require (or know anything about) the webhooks, and the whole creation of the webhooks and extension api should be hidden behind this feature gate. Then with the feature gate the controller wouldn't run anything related to the extensions at all. 

But after reading the rest of the PR and seeing how you refactored the controllers to not use the webhooks any longer, I think it should be part of this PR. You are moving the webhook registration out of the controller constructors so the constructors should no longer need to be passed in the variable. And once that change is made, then it doesn't make sense to even construct those variables outside of the feature gate block because they would not be used.

Reply:
You're correct, originally I had the feature check within the controllers which used `wh`. I removed the `wh` as a param for controllers.